src/natcap/invest/delineateit/delineateit.py

```diff
@@ -571,22 +578,37 @@ def snap_points_to_nearest_stream(points_vector_path, stream_raster_path_band,
         stream_window = stream_band.ReadAsArray(
             int(x_left), int(y_top), int(x_right - x_left),
             int(y_bottom - y_top))
+        flow_accum_array = flow_accum_band.ReadAsArray(
+            int(x_left), int(y_top), int(x_right - x_left),
+            int(y_bottom - y_top))
         row_indexes, col_indexes = numpy.nonzero(
             stream_window == 1)
-        if row_indexes.size > 0:
-            # Calculate euclidean distance for sorting
+
+        # Find the closest stream pixel that meets the distance
+        # requirement. If there is a tie, snap to the stream pixel with
+        # a higher flow accumulation value.
+        if row_indexes.size > 0:  # there are streams within the snap distance
+            # Calculate euclidean distance from the point to each stream pixel
             distance_array = numpy.hypot(
-                y_center - y_top - row_indexes,
-                x_center - x_left - col_indexes,
+                # distance along y axis from the point to each stream pixel
+                y_index - y_top - row_indexes,
+                # distance along x axis from the point to each stream pixel
+                x_index - x_left - col_indexes,
                 dtype=numpy.float32)
 
-            # Find the closest stream pixel that meets the distance
-            # requirement.
-            min_index = numpy.argmin(distance_array)
-            min_row = row_indexes[min_index]
-            min_col = col_indexes[min_index]
-            offset_row = min_row - (y_center - y_top)
-            offset_col = min_col - (x_center - x_left)
+            stream_flow_accums = flow_accum_array[row_indexes, col_indexes]
+
+            # 1d index of max value in flattened array
+            nearest_stream_index_1d = numpy.argmax(
+                (distance_array == distance_array.min()) *
+                stream_flow_accums  # weight by flow accum to break ties
+            )
```

Comment:
It does return the first occurrence. So if there are two stream pixels that are equally near to the point, and also have the same flow accumulation, it would snap to the first one (first relative to that array window).
```
>>> a
array([[1, 2, 3],
       [6, 5, 6]])
>>> numpy.argmax(a)
3
```
I'm not sure if that answers your concern?